### PR TITLE
Optionals.support

### PR DESCRIPTION
**Pull request by corinnekrych**
Link to AGIOS-354
- Add non optional support and add unit test associated
- remove Object wrapper for utilities functions, to make the code more functional looking

### CODE REVIEW

**Comment by corinnekrych** [comment]
PR on hold still investigating a failing unit test...

**Comment by corinnekrych** [comment]
@cvasilak
- I've fixed the failing test for optionals support
- also i discovered that we didn't support Array and Dictionary of primitives type: fix that + some unit tests

Do you mind to test and review it?

File: AeroGearJsonSZTests/AeroGearJsonSZTests.swift [hunk 1]
```diff
@@ -33,6 +33,132 @@ class AeroGearJsonSZTests: XCTestCase {
         super.tearDown()
     }
     
+    func testJsonToObjectModelOptionalDictionnaryOfPrimitives() {
```
Comment (cvasilak) [inline AeroGearJsonSZTests/AeroGearJsonSZTests.swift]: s/Dictionnary/Dictionary

File: AeroGearJsonSZTests/AeroGearJsonSZTests.swift [hunk 2]
```diff
@@ -33,6 +33,132 @@ class AeroGearJsonSZTests: XCTestCase {
...
+        let testSet: TestCollectionOfPrimitives = self.serializer.fromJSON(json, to: TestCollectionOfPrimitives.self)
+
+        XCTAssertTrue(testSet.dictStringString?.count == 1)
+        XCTAssertTrue(testSet.dictStringInt?.count == 1)
+        XCTAssertTrue(testSet.dictStringBool?.count == 1)
+        XCTAssertTrue(testSet.dictStringDouble?.count == 1)
+        XCTAssertTrue(testSet.dictStringFloat?.count == 1)
+    }
+    
+    func testObjectToModelOptionalDictionnaryOfPrimitives() {
```
Comment (cvasilak) [inline AeroGearJsonSZTests/AeroGearJsonSZTests.swift]: s/Dictionnary/Dictionary

File: AeroGearJsonSZTests/AeroGearJsonSZTests.swift [hunk 3]
```diff
@@ -33,6 +33,132 @@ class AeroGearJsonSZTests: XCTestCase {
...
+        XCTAssertTrue((json["arrayString"] as [String]).count == 1)
+        XCTAssertTrue((json["arrayBool"] as [Bool]).count == 1)
+        XCTAssertTrue((json["arrayInt"] as [Int]).count == 1)
+        XCTAssertTrue((json["arrayDouble"] as [Double]).count == 1)
+        XCTAssertTrue((json["arrayFloat"] as [Float]).count == 1)
+        let dict:[String] = json["arrayString"] as [String]
+        XCTAssertTrue(dict[0] == "string")
+    }
+    
+    func testJsonToObjectModelNonOptionalDictionnaryOfPrimitives() {
```
Comment (cvasilak) [inline AeroGearJsonSZTests/AeroGearJsonSZTests.swift]: s/Dictionnary/Dictionary

File: AeroGearJsonSZTests/AeroGearJsonSZTests.swift [hunk 4]
```diff
@@ -33,6 +33,132 @@ class AeroGearJsonSZTests: XCTestCase {
...
+        let testSet: TestCollectionOfPrimitivesNonOptional = self.serializer.fromJSON(json, to: TestCollectionOfPrimitivesNonOptional.self)
+        
+        XCTAssertTrue(testSet.dictStringString.count == 1)
+        XCTAssertTrue(testSet.dictStringInt.count == 1)
+        XCTAssertTrue(testSet.dictStringBool.count == 1)
+        XCTAssertTrue(testSet.dictStringDouble.count == 1)
+        XCTAssertTrue(testSet.dictStringFloat.count == 1)
+    }
+    
+    func testObjectToModelNonOptionalDictionnaryOfPrimitives() {
```
Comment (cvasilak) [inline AeroGearJsonSZTests/AeroGearJsonSZTests.swift]: s/Dictionnary/Dictionary

**Comment by cvasilak** [comment]
@corinnekrych I saw the issue now and really like the organisation that  ```conditionalFunctionCall()``` brings plus the ```TestCollectionOfPrimitivesNonOptional``` and ```TestCollectionOfPrimitives``` added test cases

feel free to merge :+1:

**Comment by cvasilak** [comment]
thank @corinnekrych :rocket: landed on https://github.com/aerogear/aerogear-ios-jsonsz/commit/a36f64f628228e7dcc692e652e084eaffd8e2246

**Pull request closed without merging**